sdk/loadtesting/load-testing-rest/test/public/testRun.spec.ts

```diff
@@ -133,58 +129,222 @@ describe("Test Run Creation", () => {
+  it("should create a load test", async () => {
+    const result = await client.path("/tests/{testId}", testId).patch({
+      contentType: "application/merge-patch+json",
+      body: {
+        displayName: "Sample Load Test",
+        description: "Sample Load Test Description",
+        loadTestConfiguration: {
+          engineInstances: 1,
+          splitAllCSVs: false,
+        },
+      },
+    });
+
+    assert.include(["200", "201"], result.status);
+  });
+
+  // Pre-req for creating a test profile run
+  it("should upload the test file with LRO", async () => {
+    const readStreamTestFile: fs.ReadStream = fs.createReadStream("./test/public/sample.jmx");
+    const fileUploadResult = await client
+      .path("/tests/{testId}/files/{fileName}", testId, "sample.jmx")
+      .put({
+        contentType: "application/octet-stream",
+        body: readStreamTestFile,
+      });
+
+    if (isUnexpected(fileUploadResult)) {
+      throw fileUploadResult.body.error;
+    }
+
+    const fileValidatePoller = await getLongRunningPoller(client, fileUploadResult);
+    await fileValidatePoller.pollUntilDone({
+      abortSignal: AbortSignal.timeout(60000), // timeout of 60 seconds
+    });
+    assert.equal(fileValidatePoller.getOperationState().status, "succeeded");
+  });
+
+  it("should create a test profile with the given test", async () => {
+    const flexFunctionsResourceId = env["LOADTESTSERVICE_FLEXFUNCTIONSRESOURCEID"] || "";
+    const testProfileDisplayName = "Sample Test Profile";
+    const testProfileDescription = "Sample Test Profile Description";
+
+    const testProfileCreationResult = await client
+      .path("/test-profiles/{testProfileId}", testProfileId)
+      .patch({
+        contentType: "application/merge-patch+json",
+        body: {
+          testId: testId,
+          displayName: testProfileDisplayName,
+          description: testProfileDescription,
+          targetResourceId: flexFunctionsResourceId,
+          targetResourceConfigurations: {
+            kind: "FunctionsFlexConsumption",
+            configurations: {
+              config1: {
+                instanceMemoryMB: 2048,
+                httpConcurrency: 20,
+              },
+              config2: {
+                instanceMemoryMB: 4096,
+                httpConcurrency: 40,
+              },
+            },
+          },
+        },
+      });
+
+    if (isUnexpected(testProfileCreationResult)) {
+      throw testProfileCreationResult.body.error;
+    }
+
+    assert.include(["200", "201"], testProfileCreationResult.status);
+  });
+
+  it("should create a test profile run and poll", async () => {
+    const testProfileRunCreationResult = await client
+      .path("/test-profile-runs/{testProfileRunId}", testProfileRunId)
+      .patch({
+        contentType: "application/merge-patch+json",
+        body: {
+          testProfileId: testProfileId,
+          displayName: "Sample Test Profile Run",
```

Comment:
Are these tests really polling the TM :o or these gets mocked ? how are they working without testId and all ?